flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -411,6 +411,25 @@
                                     .text(
                                             "Increasing this value will make the cluster more resilient against temporary resources shortages (e.g., there is more time for a failed TaskManager to be restarted), "
                                                     + "while decreasing this value reduces downtime of a job (provided that enough slots are available to still run the job).")
+                                    .linebreak()
+                                    .text(
+                                            "Setting a negative duration will disable the resource timeout: The JobManager will wait indefinitely for resources to appear.")
+                                    .build());
+
+    @Documentation.Section({
+        Documentation.Sections.EXPERT_SCHEDULING,
+        Documentation.Sections.ALL_JOB_MANAGER
+    })
+    public static final ConfigOption<Duration> RESOURCE_STABILIZATION_TIMEOUT =
```

Comment:
I think we need to document the different default values when using the reactive mode. Also for `RESOURCE_WAIT_TIMEOUT`.